tests/framework/extensions/provisioning/ssh.go

```diff
@@ -8,38 +8,85 @@ import (
 	"strconv"
 	"strings"
+	"time"
 
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	steveV1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/defaults"
+	extnodes "github.com/rancher/rancher/tests/framework/extensions/nodes"
+	"github.com/rancher/rancher/tests/framework/extensions/provisioninginput"
 	"github.com/rancher/rancher/tests/framework/pkg/nodes"
 	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 const (
-	cpuUsageVar = 100 // 100 is just a placeholder until we can determine an actual number. Even with cpu usage spiking it should not go past 100% cpu usage and previous issues concerning this were hitting around 130% and above
-	checkCPU    = "CheckCPU"
+	cpuUsageVar                                   = 100 // 100 is just a placeholder until we can determine an actual number. Even with cpu usage spiking it should not go past 100% cpu usage and previous issues concerning this were hitting around 130% and above
+	checkCPU        provisioninginput.SSHTestCase = "CheckCPU"
+	checkCPUCommand                               = "ps -C agent -o %cpu --no-header"
+	nodeReboot      provisioninginput.SSHTestCase = "NodeReboot"
+	activeState                                   = "active"
 )
 
 // This func tests the ssh tests specified in the provisioninginput config clusterSSHTests field.
 // For example CheckCPU checks the cpu usage of the cluster agent. If the usage is too high the func will return a warning.
-func CallSSHTestByName(testName string, node *nodes.Node) error {
-	switch testName {
+func CallSSHTestByName(testCase provisioninginput.SSHTestCase, node *nodes.Node, client *rancher.Client, clusterID string, clusterObject *steveV1.SteveAPIObject, steveClient *steveV1.Client) error {
+	switch testCase {
 	case checkCPU:
-		command := "ps -C agent -o %cpu --no-header"
-		output, err := node.ExecuteCommand(command)
+		output, err := node.ExecuteCommand(checkCPUCommand)
 		if err != nil {
 			return err
 		}
-		str_output := output[:strings.IndexByte(output, '\n')]
-		logrus.Info("CheckCPU test on node " + node.PublicIPAddress + " | Cluster agent cpu usage is: " + str_output + "%")
+		strOutput := output[:strings.IndexByte(output, '\n')]
+		logrus.Infof("CheckCPU test on node %s | Current cluster agent cpu usage is: %s%%", node.PublicIPAddress, strOutput)
 
-		output_int, err := strconv.ParseFloat(strings.TrimSpace(str_output), 32)
-		if output_int > cpuUsageVar {
-			logrus.Warn("Cluster agent cpu usage is too high on node" + node.PublicIPAddress + " | Current cpu usage is: " + str_output + "%")
+		outputInt, err := strconv.ParseFloat(strings.TrimSpace(strOutput), 32)
+		if outputInt > cpuUsageVar {
+			logrus.Warnf("Cluster agent cpu usage is too high on node %s | Current cluster agent cpu usage is: %s%%", node.PublicIPAddress, strOutput)
```

Comment:
Why not just have `Current cluster agent on node cpu usage is:...`? This seems a bit redundant/excessive right now.